### ISSUE TITLE
Windows Desktop App won't open

### ISSUE THREAD

**Issue by brinkmedia** [RocketChat/Rocket.Chat.Electron#851]
Just updated to the latest version, now app won't open or run. Tried to uninstall/reinstall, set to run on startup, still won't open or run, even when desktop icon is clicked.

Running on Windows 10 Pro Version 1803, 64-bit x64 processor

**Comment by kaiiiiiiiii** [comment]
A lot of my users reported that it starts minimized to the tray after the update and it took them some time to figure out ... do you may run into the same problem?

**Comment by brinkmedia** [comment]
Nope, not minimized in the tray. Just will not open at all.

**Comment by brinkmedia** [comment]
Also- in the task manager there are 4 RocketChat entries under 'Background Processes' when I click on the desktop icon and attempt to open, but the program is not open or visible anywhere. I have clicked 'End Task' and attempted to reopen the app, but it still will not open or show up anywhere.

**Comment by tassoevan** [comment]
Would you mind to upload here the file `C:\Users\<username>\AppData\Roaming\Rocket.Chat\window-state-main.json`? My bet is that your last window state was marked as hidden and tray icon is disabled.

**Comment by brinkmedia** [comment]
I don't understand what you asked me to do. Is this a file on my computer that you want me to upload here? How do I find this file?

**Comment by brinkmedia** [comment]
Never mind I found it with some help, but this window won't let me upload, says file type not supported. Next?

**Comment by tassoevan** [comment]
Indeed. I can try open it in notepad, copy its contents and paste it as comment here.

**Comment by brinkmedia** [comment]
Ok- I opened in notepad heres what it contained:

{
  "width": 1109,
  "height": 860,
  "isMaximized": false,
  "isMinimized": false,
  "isHidden": true,
  "x": 762,
  "y": 123
}

**Comment by tassoevan** [comment]
Ok, the window was really hidden. Now try to delete the same file and see what happens.

**Comment by jmason90** [comment]
Thanks @brinkmedia . Had same problem on rocketchat client minimized and hidden in system auxillary tray. Editing the json file above should resolve the issue. I found the minimized hidden window icon and opened it. Once I closed the client in an open state, that also resolved the issue.

**Comment by brinkmedia** [comment]
Success! Many thanks!

**Comment by brinkmedia** [comment]
So... this solution worked until I rebooted the computer, but the file came back and and the window won't display. Deleted again and re-booted and it opened again, but now this morning when I booted up the file is back and it won't open. How do I keep this from happening every time I restart the computer?

**Comment by jmason90** [comment]
Hi, 

For the Windows Rocketchat client, this file controls the window state 

C:\Users\Jim Mason\AppData\Roaming\Rocket.Chat\window-state-main.json
{
  "width": 992,
  "height": 596,
  "isMaximized": false,
  "isMinimized": false,
  "isHidden": false,
  "x": 268,
  "y": 104
}

When I open rocketchat after I startup my pc, I do see the full display as expected.When I minimize rocketchat to see something else, it's not accessible from the task bar easily.I have to go find it in the taskbar extensions folder to open it.
Other apps show a list of open windows when you hover over the taskbar icon ( assuming you've pinned it ). Not rocketchat client though - too bad.
Jim


   On Wednesday, September 5, 2018, 12:30:20 PM EDT, brinkmedia <notifications@github.com> wrote:  
 
 
So... this solution worked until I rebooted the computer, but the file came back and and the window won't display. Deleted again and re-booted and it opened again, but now this morning when I booted up the file is back and it won't open. How do I keep this from happening every time I restart the computer?

**Comment by brinkmedia** [comment]
Forgive my ignorance, but I don't really understand the above response... Is there a fix for this issue, and why did it just start doing this recently? Is there somewhere else I need to look if the window won't open? There is no problem when minimizing, but if I close the window I can't re-open it, and when I shut down and re-boot it will not open without deleting the file mentioned above and restarting...

**Comment by piotrkochan** [comment]
Same issue on Windows 10. Almost every day RocketChat window is not displaying and AppData/../RocketChat must be deleted.

**Comment by piotrkochan** [comment]
Maybe fix like in zeit/hyper project should be applied?
https://github.com/zeit/hyper/issues/1682
Their code: https://github.com/zeit/hyper/pull/2459/files

**Comment by tassoevan** [comment]
> Maybe fix like in zeit/hyper project should be applied?
> [zeit/hyper#1682](https://github.com/zeit/hyper/issues/1682)
> Their code: https://github.com/zeit/hyper/pull/2459/files

It is not the same issue, since there the window is visible but outside the screen bounds and here the window is actually hidden.

@piotrkochan Which app version are you running?

**Comment by piotrkochan** [comment]
I posted an issue https://github.com/RocketChat/Rocket.Chat.Electron/issues/878. Problem occurs on  2.13.2 and probably on 2.13.0

**Comment by ZsZs73** [comment]
Today I've also updated several clients on Win 7 to 2.13.3 and the window does dot show up after the update.
After killing the RC processes from the task manager I've removed the `C:\Users\<username>\AppData\Roaming\Rocket.Chat\window-state-main.json` then the window showed up.
Closing the window and launching the client again: the window does not show up...

After some testing I figured out the following.
There are two ways to exit the client:
-  click the **red X** on the window then the window closes and meanwhile the isHidden parameter turns true in the json file. (Easy to detect via keeping the json file open in Notepad++)
- via **RocketChat / Quit RocketChat** then the ishidden in the json file stays intact. (According to the Notepad++ the file gets modified, but isHidden stays false)

I hope that the above helps to catch this annoying recurring bug.

**Comment by tassoevan** [comment]
@ZsZs73 The expected behavior is to **keep Rocket.Chat running** when closing the window, persisting for futures executions that the user wanted to hide the window; **a Rocket.Chat icon must be visible on system tray/notification area** (see #854). The only way, for now, to close the client process is via `Quit Rocket.Chat` menu item. This issue is about the update procedure acting as the user manually closing the window.

However, for Windows users (specially those under Windows 7) the tray icon is not visible enough and for some Linux users there is no system tray at all. It seems that random network issues prevent the tray icon to be shown too.

**Comment by ZsZs73** [comment]
@tassoevan 
Thanks for your quick explanation.
Indeed the processes are still running after closing the window.
To solve the "tray icon is not visible enough" I did the following on Win7:

- click "Show hidden icons" on the tray
- click "Customize..."
- look for Rocket.Chat and change "Only show notifications" behavior to "Show icon and notifications"

A white RocketChat icon showing up in the tray and stays there after closing the window and most importantly clicking the RC icon in the tray shows (unhides?) the window again.
HTH someone

@tassoevan: would it be possible to default to the above tray setting?

**Comment by tassoevan** [comment]
@ZsZs73 There are some hacks for Windows, AFAIK, but this can be an abuse over user's preferences on Windows.

**Comment by jhaven** [comment]
I was having this same issue, uninstalled and reinstalled the app a few times and was about to try troubleshooting the above files as suggested.  However, during the last install, I chose to install for just my user profile instead of all users for my Windows machine.  That seemed to make it work.
